src/app/payment/paymentMethod/BoltEmbeddedPaymentMethod.tsx

```diff
@@ -9,22 +9,28 @@ const BoltEmbeddedPaymentMethod: FunctionComponent<HostedPaymentMethodProps> = (
     method,
     ...rest
 }) => {
+    const [ showCreateAccountCheckbox, setShowCreateAccountCheckbox ] = useState(false);
+
     const boltEmbeddedContainerId = 'bolt-embedded';
 
     const initializeBoltPayment = useCallback(options => initializePayment({
             ...options,
             bolt: {
                 containerId: boltEmbeddedContainerId,
                 useBigCommerceCheckout: true,
+                onPaymentSelect: (hasBoltAccount: boolean) => {
+                    setShowCreateAccountCheckbox(!hasBoltAccount);
+                },
             },
         }
     ), [initializePayment, boltEmbeddedContainerId]);
 
     const renderCustomPaymentForm = useCallback(() => (
```

Comment:
Should this be `useCallback` or `useMemo`

Reply:
Looked at this further and how the prop is implement it expects a function so useCallback is justified here